packages/fx-core/src/common/deps-checker/depsManager.ts

```diff
@@ -63,7 +65,8 @@ export class DepsManager {
    */
   public async ensureDependencies(
     dependencies: DepsType[],
-    { fastFail = true, doctor = false }: DepsOptions
+    { fastFail = true, doctor = false }: DepsOptions,
+    progressBar?: IProgressHandler
```

Comment:
DepsManager shouldn't hold progressBar cause it is related to UI.
We can solve it by this way: 
```
async function checkDependencies() {
     xxx
    for(const dep of nonNodeDeps) {
      doctor.log(xxx); // one of my PR will contains this line, and also remove doctor log in DepsManager.
      await progressBar?.next(DepsCheckerProgressMessage[dep]); 
        depsManager.ensureDependencies([dep], {fastFail: false, doctor: true});
    }
   xxx
}
```

Reply:
Shall we move `this.sortBySequence(dependencies, DepsManager.depsOrders);` into VSC?

Reply:
@XiaofuHuang just do it in `getActiveDependencies()`